src/pgmoneta_ext/lib.c

```diff
@@ -52,3 +61,37 @@ pgmoneta_ext_version(PG_FUNCTION_ARGS)
 
    PG_RETURN_DATUM(version);
 }
+
+Datum
+pgmoneta_ext_switch_wal(PG_FUNCTION_ARGS)
+{
+   Datum result;
+   XLogRecPtr recptr;
+   char str_res[1024];
+   bool is_superuser;
+   Oid roleid;
+
+   // Get the current role's ID
+   roleid = GetUserId();
+
+   is_superuser = pgmoneta_ext_check_privilege(roleid);
+
+   if (!is_superuser)
+   {
+      elog(LOG, "pgmoneta_ext_switch_wal: current role is not a superuser");
+
+      memset(&str_res, 0, sizeof(str_res));
+      snprintf(&str_res[0], sizeof(str_res), "%s", "false | NULL");
+   }
+   else
+   {
+      // Call the RequestXLogSwitch function with mark_unimportant as false
+      recptr = RequestXLogSwitch(false);
+
+      snprintf(str_res, sizeof(str_res), "true | %X/%X", (uint32) (recptr >> 32), (uint32) recptr);
```

Comment:
I think what @jesperpedersen meant was you should return two cols instead of a formatted string

Reply:
Okay, got it.

I attempted to create a new type that includes two columns. Is this an effective way to achieve this?

``` sql
CREATE TYPE pgmoneta_ext_result AS (
   success boolean,
   str_res text
);
```

Reply:
Perhaps you could try https://www.postgresql.org/docs/current/xfunc-c.html#XFUNC-C-RETURNING-ROWS. Checkout `HeapTupleGetDatum` and  `heap_form_tuple`. I think there is a way to construct the tuple descriptor - check out `TupleDescInitEntry`. Let me know if you have better ideas.

Reply:
Take a look at other examples like https://github.com/pgexporter/pgexporter_ext/blob/main/src/pgexporter_ext/lib.c#L444 on how to build a Datum with multiple columns